subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -493,26 +510,37 @@ int bt_mesh_adv_enable(void)
 		}
 	}
 
+#if defined(CONFIG_BT_MESH_WORKQ_MESH)
+	k_work_queue_init(&bt_mesh_workq);
+	k_work_queue_start(&bt_mesh_workq, thread_stack, CONFIG_BT_MESH_ADV_STACK_SIZE,
+			   K_PRIO_COOP(CONFIG_BT_MESH_ADV_PRIO), NULL);
+	k_thread_name_set(&bt_mesh_workq.thread, "BT MESH WQ");
+#endif
+
 	return 0;
 }
 
 int bt_mesh_adv_disable(void)
 {
-	int err;
 	struct k_work_sync sync;
+	int err = 0;
 
 	for (int i = 0; i < ARRAY_SIZE(advs); i++) {
 		atomic_set_bit(advs[i].flags, ADV_FLAG_SUSPENDING);
 
-		if (k_current_get() != &k_sys_work_q.thread ||
-		    (k_work_busy_get(&advs[i].work) & K_WORK_RUNNING) == 0) {
+		if (!(k_work_busy_get(&advs[i].work) & K_WORK_RUNNING)
+#if defined(CONFIG_BT_MESH_WORKQ_SYS)
+		    || (k_current_get() != &k_sys_work_q.thread)) {
+#else  /* CONFIG_BT_MESH_WORKQ_MESH */
+		    || (k_current_get() != &bt_mesh_workq.thread)) {
+#endif /* CONFIG_BT_MESH_WORKQ_SYS */
```

Comment:
I think this could be made slightly more readable with something like:
```c
#if defined(CONFIG_BT_MESH_WORKQ_SYS)
#define MESH_WORKQ_THREAD &k_sys_work_q.thread
#else
#define MESH_WORKQ_THREAD &bt_mesh_workq.thread
#endif
...
	if (!k_work_busy_get(&advs[i].work) & K_WORK_RUNNING) ||
	    k_current_get() != MESH_WORKQ_THREAD) {
		k_work_flush(&advs[i].work, &sync);
	}
```